src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,121 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private final StorageDaoImpl storageDao = new StorageDaoImpl();
+    private final RegistrationServiceImpl registrationService = new RegistrationServiceImpl();
+
+    @BeforeEach
+    void setUp() {
+        storageDao.add(new User(1L, "pavlo_ivanets", "458984", 18));
+        storageDao.add(new User(2L, "katerynka", "fks2045", 24));
+    }
+
+    @Test
+    void register_LoginCorrectLength_Ok() {
+        User user1 = new User(4L, "kate43", "tyo7654", 25);
+        User user2 = new User(4L, "kate4305", "tyo7654", 25);
+        User actual1 = storageDao.add(user1);
+        User actual2 = storageDao.add(user2);
```

Comment:
The `register_LoginCorrectLength_Ok` test should use `registrationService.register()` instead of `storageDao.add()` to ensure the registration logic is tested, including validation checks.